amanuensis/resources/filter_sets/__init__.py

```diff
@@ -0,0 +1,263 @@
+            number_options = ("number", ["number"])
+            if es_to_dd_map[filter]["type"] in text_options:
+                
+                if any(not isinstance(value, str) for value in selected_values):
+                    logger.error(f"The filter-set {filter_set_name} contains the value {selected_values} for the filter {filter} which is not a valid value for the filter {filter} in the data dictionary")
+                    return False
+
+            elif es_to_dd_map[filter]["type"] in number_options:
+                
+                if any(not isinstance(value, int) and not isinstance(value, float) for value in selected_values):
+                    logger.error(f"The filter-set {filter_set_name} contains the value {selected_values} for the filter {filter} which is not a valid value for the filter {filter} in the data dictionary")
+                    return False
+            
+            else:
+                logger.error(f"The filter-set {filter_set_name} contains the filter {filter} which has an unsupported type: {es_to_dd_map[filter]['type']}")
+                return False
+        
+        
+    logger.info(f"The filter {filter_set_name} is valid for the data dictionary")   
+    return True   
+
+
+def check_filter_sets(session, es_to_dd_map_file_name="es_to_dd_map.json", portal_config_file_name="gitops.json"):
+
+    es_to_dd_map = _load_data_files(es_to_dd_map_file_name)
+    
+    portal_config = _load_data_files(portal_config_file_name)
+
+    explorer_selectable_values = _get_selectable_filters_from_data_portal(portal_config)
+
+    filter_sets = get_filter_sets(session, filter_by_active=False, filter_by_source_type=False)
+
+    for filter_set in filter_sets:
+
+        graphql_object = filter_set.graphql_object
+        filter_set_name = filter_set.name
+        filter_source_internal_id = filter_set.filter_source_internal_id
+        filter_source = filter_set.filter_source
+
+        separator_length = len(f"CHECKING FILTER SET: {filter_set_name} id: {filter_set.id}")
+        logger.info("*" * separator_length)
+        logger.info(f"CHECKING FILTER SET: {filter_set_name} id: {filter_set.id}")
+        logger.info("*" * separator_length)
+        logger.info(f"graphql_object: {graphql_object}")
+
+        is_valid = False
+            
+        if graphql_object is None:
+
+            logger.info(f"Filter-set {filter_set_name} has no graphql_object, marking invalid")
+        
+        else:
+            
+            extracted_values_from_filter_set = _extract_selected_values_from_filter_set(graphql_object)
+        
+            if extracted_values_from_filter_set is not False:
+                
+                if not filter_source_internal_id and filter_source == "manual":
+                    logger.info(f"Filter-set {filter_set_name} is a manual filter-set, skipping validation against data portal explorer values")
```

Comment:
This will set to inactive all the manual filters. I would keep them as they are. What is your reason for disabling them?

Reply:
This just check if the filter-set is manual and if so it skips the part where it checks against the portal selectable values since there is no explorer id and just checks against the DD values. But I did notice a bug where if invalid for the portal values but valid for DD it marks it as valid. which I added the changes for.